src/pages/Swap/Swap.tsx

```diff
@@ -33,8 +50,80 @@ export const Swap: React.FC<SwapProp> = ({
     if (!isConnected) {
       return alert('User must enable this website to proceed');
     }
+    handleMarkets();
+  };
+
+  const amountIsPositive = (x: any): boolean => {
+    if (!Number.isNaN(x) && Number(x) > 0) {
+      return true;
+    }
+
+    return false;
+  };
+
+  const handleSendAmountChange = async (
+    evt: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+  ) => {
+    if (isPreviewing) return;
+
+    const value = evt.target.value;
+    setAmountToBeSent(value);
+
+    if (amountIsPositive(value) && selectError === '') {
+      setIsPreviewing(true);
+
+      const amount = Number(value);
+
+      const direction = marketDirection(checkedCoin);
+
+      try {
+        const toRecieve = await onSendAmountChange(
+          amount,
+          direction ? TradeType.SELL : TradeType.BUY,
+          Market.L_BTC,
```

Comment:
How you can know the asset the user is entering the amount is always L_BTC? this should flip based on what the dropdown select is

Reply:
fixed this